engine/class_modules/sc_death_knight.cpp

```diff
@@ -4552,7 +4573,7 @@ struct death_and_decay_t : public death_and_decay_base_t
 struct defile_t : public death_and_decay_base_t
 {
   defile_t( death_knight_t* p, const std::string& options_str ) :
-    death_and_decay_base_t( p, "defile", p -> talent.defile )
+    death_and_decay_base_t( p, p -> covenant.deaths_due -> ok() ? "deaths_due" : "defile", p -> talent.defile )
```

Comment:
Is the spell named Defile or Death's Due with the covenant active?

Reply:
it's named defile with the covenant active.  Casting drops death's due graphic, with the death's due buffs.  It shows up in details as defile as well.  if I remember right, when they first put the spell in, it has no interaction with defile.

Reply:
Then i think the name here should be Defile as well, to better reflect what the users would see on logs/details.

Reply:
I can go either way on this, but originally I had gone in this direction for the naming due to how it looks in the html report.

![image](https://user-images.githubusercontent.com/3732166/99033490-d7413e80-252f-11eb-93b3-4419b332bea5.png)


Text report however only reports on the one name:  deaths_due
![image](https://user-images.githubusercontent.com/3732166/99033517-ee802c00-252f-11eb-8498-5890c3566ea4.png)

I felt the html report was more clear for defile this way, and left no questions about it,  at the expense of the text report.

Reply:
Yeah the HTML report will fetch tooltip for the spell ID and show the name associated with it regardless of the actual name in the c++ code. Whether the spell and its damage component are called defile, death's due, or something else is up to you but I think calling it Defile would at least fit what it's doing ingame (just a Defile spell and damage, with death's due mechanics band-aided on it if I got it right).

Reply:
Yes, defile spell with death's due mechanics band-aided on.  Updated to report it as defile.